test_tf2/test/test_message_filter.cpp

```diff
@@ -179,6 +179,47 @@ TEST(MessageFilter, postTransforms)
 
   EXPECT_EQ(1, n.count_);
 }
+
+TEST(MessageFilter, multithread)
+{
+  const int messages = 30;
+  const int buffer_size = messages;
+  auto node = rclcpp::Node::make_shared("tf2_ros_message_filter");
+  auto create_timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    node->get_node_base_interface(),
+    node->get_node_timers_interface());
+
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+
+  builtin_interfaces::msg::Time stamp = tf2_ros::toMsg(tf2::timeFromSec(1));
+
+  std::shared_ptr<geometry_msgs::msg::PointStamped> msg = std::make_shared<geometry_msgs::msg::PointStamped>();
+  msg->header.stamp = stamp;
+  msg->header.frame_id = "frame2";
+
+  for (int i = 0; i < 50; i++) {
+    tf2_ros::Buffer buffer(clock);
```

Comment:
@ymd-stella consider pulling the buffer out of the loop.

Reply:
Fixed it.